internal/transform/transform.go

```diff
@@ -453,3 +454,22 @@ func walk(doc *astro.Node, cb func(*astro.Node)) {
 	}
 	f(doc)
 }
+
+// This function merges the values of `class=""` and `class:list=""` in `class:list`
+func mergeClassList(doc *astro.Node, n *astro.Node, opts *TransformOptions) {
+	var classListAttrValue string
+	var classListAttrIndex int
+	var classAttrValue string
+	for i, a := range n.Attr {
+		if a.Key == "class:list" {
+			classListAttrValue = a.Val
+			classListAttrIndex = i
+		}
+		if a.Key == "class" {
+			classAttrValue = a.Val
+		}
+	}
+	if classListAttrIndex > 0 {
```

Comment:
Seems like we could remove the `class` attribute if we find one, since it will be merged into the `class:list` declaration anyway? That would ensure we generate valid HTML with only one `class` attribute.

Not the cleanest thing to do in Go. You basically have to create a new array with that element removed.

```go
func remove(slice []int, s int) []int {
    return append(slice[:s], slice[s+1:]...)
}
```